include/aws/testing/aws_test_allocators.h

```diff
@@ -0,0 +1,97 @@
+#ifndef AWS_TESTING_AWS_TEST_ALLOCATORS_H
+#define AWS_TESTING_AWS_TEST_ALLOCATORS_H
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/testing/aws_test_harness.h>
+
+/** \file
+ * Alternate allocators for use in testing.
+ */
+
+/**
+ * Timebomb allocator fakes running out of memory after then Nth allocation.
+ * Once this allocator starts failing, it never succeeds, even if memory is released.
+ * Wraps an existing allocator.
+ */
+struct aws_timebomb_impl {
```

Comment:
Would it be interresting to potentially track how many bytes were requested, and give a max # of bytes?

Reply:
That would be interesting.
But that would be a job for a different allocator.

This is the timebomb allocator.